$Id: 011a_recognize_glibc_2.3.2_locale_names.diff 1064 2006-01-11 12:26:45Z ender $

Major housecleaning of X11's locale support.

* Add all locales supported by version 2.3.2 of the GNU C Library except
  zh_CN.GB18030, sq_AL.ISO-8859-1, and tg_TJ.KOI8-T.  X has no support for
  the GB18030 and KOI8-T character sets, and glibc appears to disagree with
  most of the world about which character set Albanians use (see, for
  example, iso_8859-2(7)).  This baka gaijin has learned his lesson about
  trying to report l10n bugs in glibc, so I'll let someone else do the
  jousting.  :-P

* Synchronize compose.dir with locale.dir wherever possible (i.e., wherever
  compose tables exist) so that as many supported locales as possible get a
  Compose map.  Comment out references to compose tables that don't
  actually exist.

* Disable support for iu_CA.NUNACOM-8.  X11 doesn't have any support for
  the NUNACOM-8 character set.  (iu_CA.UTF-8 is still supported.)

* Disable support for locales that are neither supported by the GNU C
  Library, nor use standard ISO 639 language or 3166 territory codes.  This
  includes Esperanto (eo_EO, eo_XX), and what I presume is Pennsylvania
  "Dutch" (pd_DE, pd_US).

* Turn off intrusion of Nynorsk into Nyanja's ISO 639 space.  (Nynorsk is
  nn, not ny.)

* Retain support for a3 as an alias for az (Azerbaijani), though I'm dubious
  about it.

* Turn off support for the bogus (and ancient) FreeBSD-ism of using
  "lt_LN.ISO8859-1" as a "generic" Latin-1 locale.

* Remove bogus "ee" locales for Estonian (et).

* Remove ISO 8859-15 support for languages that don't use that character
  set according to glibc.

* Remove @euro modifier support from locales that don't use that currency
  (i.e., countries not in the Euro zone).

* Ensure we support the @euro modifier for all locales that use it
  (according to glibc), even if the codeset is explicitly specified.  See
  Debian #208137.

* Alias "fi" and "fi_FI" to "fi_FI.ISO8559-1", not "fi_FI.ISO8859-15", for
  compatibility with the GNU C Library.  See Debian #266834.

* Rewrite some comments to be in grammatical English.

This patch by Branden Robinson.
Major "forward port" by Nathanael Nerode.

Partially submitted upstream.  This is so large I don't expect it to all go in at once,
but any bit would help.  --Nathanael

Index: libx11/nls/compose.dir.pre
===================================================================
--- libx11.orig/nls/compose.dir.pre
+++ libx11/nls/compose.dir.pre
@@ -1,14 +1,22 @@
 XCOMM $Xorg: compose.dir,v 1.3 2000/08/17 19:46:48 cpqbld Exp $
 XCOMM
-XCOMM	This file contains compose table file name.
-XCOMM	The first word is the compose table file name and
-XCOMM	the second word is full locale name.
+XCOMM	This file contains compose table file names.  The first word is the
+XCOMM	compose table file name and the second word is the full locale
+XCOMM	name.
 XCOMM
 XCOMM $XFree86: xc/nls/compose.dir,v 1.24 2003/08/06 14:04:00 eich Exp $
 XCOMM
+XCOMM Note that no Compose tables exist for ISO 8859-11, Microsoft code
+XCOMM pages, Fujitsu's S90 or U90 character sets, or TIS620 yet.
+XCOMM
 iso8859-1/Compose:		C
+XCOMM a3 is not an ISO 639 language code, but in Cyrillic, "Z" looks like "3".
+koi8-c/Compose:			a3_AZ.KOI8-C
+iso8859-1/Compose:		aa_DJ.ISO8859-1
 iso8859-1/Compose:		af_ZA.ISO8859-1
+iso8859-15/Compose:		an_ES.ISO8859-15
 iso8859-6/Compose:		ar_AA.ISO8859-6
+iso8859-6/Compose:		ar_AE.ISO8859-6
 iso8859-6/Compose:		ar_BH.ISO8859-6
 iso8859-6/Compose:		ar_DZ.ISO8859-6
 iso8859-6/Compose:		ar_EG.ISO8859-6
@@ -21,13 +29,17 @@
 iso8859-6/Compose:		ar_OM.ISO8859-6
 iso8859-6/Compose:		ar_QA.ISO8859-6
 iso8859-6/Compose:		ar_SA.ISO8859-6
+iso8859-6/Compose:     ar_SD.ISO8859-6
+iso8859-6/Compose:     ar_SY.ISO8859-6
 iso8859-6/Compose:		ar_TN.ISO8859-6
 iso8859-6/Compose:		ar_YE.ISO8859-6
 iso8859-9e/Compose:		az_AZ.ISO8859-9E
-koi8-c/Compose:			a3_AZ.KOI8-C
-microsoft-cp1251/Compose:	be_BY.CP1251
+koi8-c/Compose:			az_AZ.KOI8-C
+iso8859-5/Compose:     be_BY.ISO8859-5
+XCOMM microsoft-cp1251/Compose:    be_BY.CP1251
 iso8859-5/Compose:		bg_BG.ISO8859-5
-microsoft-cp1251/Compose:	bg_BG.CP1251
+XCOMM microsoft-cp1251/Compose:    bg_BG.CP1251
+koi8-r/Compose:            bg_BG.KOI8-R
 iso8859-1/Compose:		br_FR.ISO8859-1
 iso8859-14/Compose:		br_FR.ISO8859-14
 iso8859-15/Compose:		br_FR.ISO8859-15
@@ -42,6 +54,8 @@
 iso8859-15/Compose:		da_DK.ISO8859-15
 iso8859-1/Compose:		de_AT.ISO8859-1
 iso8859-15/Compose:		de_AT.ISO8859-15
+iso8859-1/Compose:		de_BE.ISO8859-1
+iso8859-15/Compose:		de_BE.ISO8859-15
 iso8859-1/Compose:		de_CH.ISO8859-1
 iso8859-15/Compose:		de_CH.ISO8859-15
 iso8859-1/Compose:		de_DE.ISO8859-1
@@ -51,15 +65,25 @@
 iso8859-1/Compose:		de_LU.ISO8859-1
 iso8859-15/Compose:		de_LU.ISO8859-15
 iso8859-7/Compose:		el_GR.ISO8859-7
+iso8859-15/Compose:		el_GR.ISO8859-15
 iso8859-1/Compose:		en_AU.ISO8859-1
+iso8859-1/Compose:		en_BE.ISO8859-1
+iso8859-15/Compose:		en_BE.ISO8859-15
+iso8859-1/Compose:		en_BW.ISO8859-1
 iso8859-1/Compose:		en_BZ.ISO8859-1
 iso8859-1/Compose:		en_CA.ISO8859-1
+iso8859-1/Compose:		en_DK.ISO8859-1
 iso8859-1/Compose:		en_GB.ISO8859-1
 iso8859-15/Compose:		en_GB.ISO8859-15
+iso8859-1/Compose:		en_HK.ISO8859-1
 iso8859-1/Compose:		en_IE.ISO8859-1
 iso8859-15/Compose:		en_IE.ISO8859-15
+iso8859-1/Compose:		en_IN.ISO8859-1
+iso8859-15/Compose:		en_IN.ISO8859-15
 iso8859-1/Compose:		en_JM.ISO8859-1
 iso8859-1/Compose:		en_NZ.ISO8859-1
+iso8859-1/Compose:		en_PH.ISO8859-1
+iso8859-1/Compose:		en_SG.ISO8859-1
 iso8859-1/Compose:		en_TT.ISO8859-1
 iso8859-1/Compose:		en_UK.ISO8859-1
 iso8859-15/Compose:		en_UK.ISO8859-15
@@ -67,7 +91,11 @@
 iso8859-15/Compose:		en_US.ISO8859-15
 iso8859-1/Compose:		en_ZA.ISO8859-1
 iso8859-15/Compose:		en_ZA.ISO8859-15
-iso8859-3/Compose:		eo_XX.ISO8859-3
+iso8859-1/Compose:		en_ZW.ISO8859-1
+XCOMM Esperato (eo) is not supported by the GNU C Library and neither the
+XCOMM EO nor XX territories exist in ISO 3166.
+XCOMM iso8859-3/Compose:	eo_EO.ISO8859-3
+XCOMM iso8859-3/Compose:	eo_XX.ISO8859-3
 iso8859-1/Compose:		es_AR.ISO8859-1
 iso8859-1/Compose:		es_BO.ISO8859-1
 iso8859-1/Compose:		es_CL.ISO8859-1
@@ -86,6 +114,7 @@
 iso8859-1/Compose:		es_PR.ISO8859-1
 iso8859-1/Compose:		es_PY.ISO8859-1
 iso8859-1/Compose:		es_SV.ISO8859-1
+iso8859-1/Compose:		es_US.ISO8859-1
 iso8859-1/Compose:		es_UY.ISO8859-1
 iso8859-1/Compose:		es_VE.ISO8859-1
 iso8859-1/Compose:		et_EE.ISO8859-1
@@ -94,6 +123,8 @@
 iso8859-15/Compose:		et_EE.ISO8859-15
 iso8859-1/Compose:		eu_ES.ISO8859-1
 iso8859-15/Compose:		eu_ES.ISO8859-15
+iso8859-1/Compose:		eu_FR.ISO8859-1
+iso8859-15/Compose:		eu_FR.ISO8859-15
 isiri-3342/Compose:		fa_IR.ISIRI-3342
 iso8859-1/Compose:		fi_FI.ISO8859-1
 iso8859-15/Compose:		fi_FI.ISO8859-15
@@ -121,17 +152,15 @@
 iso8859-14/Compose:		gv_GB.ISO8859-14
 iso8859-15/Compose:		gv_GB.ISO8859-15
 iso8859-8/Compose:		he_IL.ISO8859-8
-microsoft-cp1255/Compose:	he_IL.CP1255
+XCOMM microsoft-cp1255/Compose:	he_IL.CP1255
 iscii-dev/Compose:		hi_IN.ISCII-DEV
 iso8859-2/Compose:		hr_HR.ISO8859-2
 iso8859-2/Compose:		hu_HU.ISO8859-2
 armscii-8/Compose:		hy_AM.ARMSCII-8
 iso8859-1/Compose:		id_ID.ISO8859-1
-iso8859-15/Compose:		id_ID.ISO8859-15
 iso8859-1/Compose:		is_IS.ISO8859-1
 iso8859-15/Compose:		is_IS.ISO8859-15
 iso8859-1/Compose:		it_CH.ISO8859-1
-iso8859-15/Compose:		it_CH.ISO8859-15
 iso8859-1/Compose:		it_IT.ISO8859-1
 iso8859-15/Compose:		it_IT.ISO8859-15
 ja/Compose:			ja_JP.eucJP
@@ -140,14 +169,20 @@
 georgian-academy/Compose:	ka_GE.GEORGIAN-ACADEMY
 georgian-ps/Compose:		ka_GE.GEORGIAN-PS
 iso8859-1/Compose:		kl_GL.ISO8859-1
-iso8859-15/Compose:		kl_GL.ISO8859-15
 iso8859-1/Compose:		kw_GB.ISO8859-1
 iso8859-14/Compose:		kw_GB.ISO8859-14
 iso8859-15/Compose:		kw_GB.ISO8859-15
+iso8859-10/Compose:		lg_UG.ISO8859-10
 ko/Compose:			ko_KR.eucKR
 ibm-cp1133/Compose:		lo_LA.IBM-CP1133
 mulelao-1/Compose:		lo_LA.MULELAO-1
-iso8859-1/Compose:		lt_LN.ISO8859-1
+#if defined(__FreeBSD__)
+XCOMM most locales in FreeBSD 2.1.[56] do not work, allow use of FreeBSD's
+XCOMM generic latin-1 locale name. N.B. "lt" is a bad code to use for a
+XCOMM generic Latin-1 locale anyway because it is the country code for
+XCOMM Lithuania, which by default uses Latin-7 (ISO-8859-13).
+XCOMM iso8859-1/Compose:	lt_LN.ISO8859-1
+#endif
 iso8859-4/Compose:		lt_LT.ISO8859-4
 iso8859-13/Compose:		lt_LT.ISO8859-13
 iso8859-4/Compose:		lv_LV.ISO8859-4
@@ -155,7 +190,7 @@
 iso8859-13/Compose:		mi_NZ.ISO8859-13
 iso8859-5/Compose:		mk_MK.ISO8859-5
 XCOMM mk_MK.UTF-8/Compose:		mk_MK.UTF-8
-microsoft-cp1251/Compose:	mk_MK.CP1251
+XCOMM microsoft-cp1251/Compose:	mk_MK.CP1251
 iso8859-1/Compose:		ms_MY.ISO8859-1
 iso8859-3/Compose:		mt_MT.ISO8859-3
 iso8859-1/Compose:		nl_BE.ISO8859-1
@@ -174,25 +209,32 @@
 iso8859-15/Compose:		ny_NO.ISO8859-15
 iso8859-1/Compose:		oc_FR.ISO8859-1
 iso8859-15/Compose:		oc_FR.ISO8859-15
-iso8859-1/Compose:		pd_DE.ISO8859-1
-iso8859-1/Compose:		pd_US.ISO8859-1
+iso8859-1/Compose:     om_KE.ISO8859-1
+XCOMM Presumably, "pd" is "Pennsylvania 'Dutch'", a U.S. dialect of
+German.
+XCOMM This is not an ISO language code defined for it, and the GNU C
+Library
+XCOMM doesn't support it.
+XCOMM iso8859-1/Compose:   pd_DE.ISO8859-1
+XCOMM iso8859-1/Compose:   pd_US.ISO8859-1
 iso8859-1/Compose:		ph_PH.ISO8859-1
 iso8859-2/Compose:		pl_PL.ISO8859-2
 iso8859-1/Compose:		pp_AN.ISO8859-1
 iso8859-1/Compose:		pt_BR.ISO8859-1
-iso8859-15/Compose:		pt_BR.ISO8859-15
 iso8859-1/Compose:		pt_PT.ISO8859-1
 iso8859-15/Compose:		pt_PT.ISO8859-15
 iso8859-2/Compose:		ro_RO.ISO8859-2
+XCOMM microsoft-cp1251/Compose:    ru_RU.CP1251
 koi8-r/Compose:			ru_RU.KOI8-R
-iso8859-5/Compose:		ru_RU.ISO8859-5
-microsoft-cp1251/Compose:	ru_RU.CP1251
+XCOMM microsoft-cp1251/Compose:    ru_UA.CP1251
 koi8-u/Compose:			ru_UA.KOI8-U
 iso8859-1/Compose:		rw_RW.ISO8859-1
 iso8859-2/Compose:		sh_YU.ISO8859-2
 iso8859-2/Compose:		sk_SK.ISO8859-2
-iso8859-2/Compose:		sl_CS.ISO8859-2
 iso8859-2/Compose:		sl_SI.ISO8859-2
+iso8859-1/Compose:      so_DJ.ISO8859-1
+iso8859-1/Compose:      so_KE.ISO8859-1
+iso8859-1/Compose:      so_SO.ISO8859-1
 iso8859-5/Compose:		sp_YU.ISO8859-5
 iso8859-2/Compose:		sq_AL.ISO8859-2
 iso8859-2/Compose:		sr_CS.ISO8859-2
@@ -200,6 +242,7 @@
 iso8859-2/Compose:		sr_SP.ISO8859-2
 iso8859-2/Compose:		sr_YU.ISO8859-2
 iso8859-5/Compose:		sr_YU.ISO8859-5
+XCOMM microsoft-cp1251/Compose:    sr_YU.CP1251
 iso8859-1/Compose:		ss_ZA.ISO8859-1
 iso8859-1/Compose:		st_ZA.ISO8859-1
 iso8859-1/Compose:		sv_FI.ISO8859-1
@@ -208,38 +251,59 @@
 iso8859-15/Compose:		sv_SE.ISO8859-15
 tscii-0/Compose:		ta_IN.TSCII-0
 koi8-c/Compose:			tg_TJ.KOI8-C
+XCOMM There was once a blank th_TH Compose map, but it was removed.  See
+XCOMM <URL: http://linux.thai.net/~thep/thaisupp/ >.
+XCOMM iso8859-11/Compose:  th_TH.ISO8859-11
+XCOMM th_TH/Compose:       th_TH.TIS620
 iso8859-1/Compose:		tl_PH.ISO8859-1
 iso8859-15/Compose:		tn_ZA.ISO8859-15
 iso8859-9/Compose:		tr_TR.ISO8859-9
 iso8859-1/Compose:		ts_ZA.ISO8859-1
 tatar-cyr/Compose:		tt_RU.TATAR-CYR
 koi8-c/Compose:			tt_RU.KOI8-C
+iso8859-5/Compose:     uk_UA.ISO8859-5
 koi8-u/Compose:			uk_UA.KOI8-U
-microsoft-cp1256/Compose:	ur_PK.CP1256
+XCOMM microsoft-cp1251/Compose:    uk_UA.CP1251
+XCOMM microsoft-cp1256/Compose:    ur_PK.CP1256
+iso8859-1/Compose:     uz_UZ.ISO8859-1
 vi_VN.viscii/Compose:		vi_VN.VISCII
 vi_VN.tcvn/Compose:		vi_VN.TCVN
 iso8859-1/Compose:		wa_BE.ISO8859-1
+iso8859-15/Compose:        wa_BE.ISO8859-15
 iso8859-1/Compose:		xh_ZA.ISO8859-1
+XCOMM microsoft-cp1255/Compose:    yi_US.CP1255
 microsoft-cp1255/Compose:	yi_US.CP1256
 zh_CN/Compose:			zh_CN.eucCN
-zh_CN/Compose:			zh_CN.GB2312
-zh_CN.gbk/Compose:		zh_CN.GBK
+zh_CN/Compose:         zh_CN.gb2312
+zh_CN.gbk/Compose:     zh_CN.gbk
+zh_HK.big5/Compose:        zh_HK.big5
+zh_HK.big5hkscs/Compose:   zh_HK.big5-hkscs
+zh_HK.big5hkscs/Compose:   zh_HK.Big5-HKSCS
+zh_HK.big5hkscs/Compose:   zh_HK.big5hkscs
+zh_HK.big5hkscs/Compose:   zh_HK.Big5HKSCS
+zh_CN/Compose:         zh_SG.gb2312
+zh_CN.gbk/Compose:     zh_SG.gbk
 zh_CN.gb18030/Compose:		zh_CN.GB18030
 zh_HK.big5/Compose:		zh_HK.big5
-zh_HK.big5/Compose:		zh_HK.Big5
 zh_HK.big5hkscs/Compose:	zh_HK.big5hkscs
 zh_HK.big5hkscs/Compose:	zh_HK.Big5HKSCS
 zh_TW.big5/Compose:		zh_TW.big5
 zh_TW.big5/Compose:		zh_TW.Big5
 zh_TW/Compose:			zh_TW.eucTW
 iso8859-1/Compose:		zu_ZA.ISO8859-1
-XCOMM
-XCOMM
+XCOMM Note: The UTF-8 locales don't work correctly yet. Work in progress.
+XCOMM a3 is not an ISO 639 language code, but in Cyrillic, "Z" looks like "3".
+en_US.UTF-8/Compose:       a3_AZ.UTF-8
+en_US.UTF-8/Compose:       aa_ER.UTF-8
+en_US.UTF-8/Compose:       aa_ET.UTF-8
 en_US.UTF-8/Compose:		af_ZA.UTF-8
+en_US.UTF-8/Compose:       am_ET.UTF-8
 en_US.UTF-8/Compose:		ar_AA.UTF-8
+en_US.UTF-8/Compose:       ar_AE.UTF-8
 en_US.UTF-8/Compose:		ar_BH.UTF-8
 en_US.UTF-8/Compose:		ar_DZ.UTF-8
 en_US.UTF-8/Compose:		ar_EG.UTF-8
+en_US.UTF-8/Compose:       ar_IN.UTF-8
 en_US.UTF-8/Compose:		ar_IQ.UTF-8
 en_US.UTF-8/Compose:		ar_JO.UTF-8
 en_US.UTF-8/Compose:		ar_KW.UTF-8
@@ -249,15 +313,18 @@
 en_US.UTF-8/Compose:		ar_OM.UTF-8
 en_US.UTF-8/Compose:		ar_QA.UTF-8
 en_US.UTF-8/Compose:		ar_SA.UTF-8
+en_US.UTF-8/Compose:       ar_SD.UTF-8
+en_US.UTF-8/Compose:       ar_SY.UTF-8
 en_US.UTF-8/Compose:		ar_TN.UTF-8
 en_US.UTF-8/Compose:		ar_YE.UTF-8
 en_US.UTF-8/Compose:		bs_BA.UTF-8
 en_US.UTF-8/Compose:		az_AZ.UTF-8
-en_US.UTF-8/Compose:		a3_AZ.UTF-8
 en_US.UTF-8/Compose:		be_BY.UTF-8
 en_US.UTF-8/Compose:		bg_BG.UTF-8
+en_US.UTF-8/Compose:       bn_BD.UTF-8
 en_US.UTF-8/Compose:		bn_IN.UTF-8
 en_US.UTF-8/Compose:		br_FR.UTF-8
+en_US.UTF-8/Compose:       byn_ER.UTF-8
 en_US.UTF-8/Compose:		ca_ES.UTF-8
 en_US.UTF-8/Compose:		cs_CZ.UTF-8
 en_US.UTF-8/Compose:		cy_GB.UTF-8
@@ -271,17 +338,28 @@
 en_US.UTF-8/Compose:		de_LU.UTF-8
 el_GR.UTF-8/Compose:		el_GR.UTF-8
 en_US.UTF-8/Compose:		en_AU.UTF-8
+en_US.UTF-8/Compose:		en_BE.UTF-8
+en_US.UTF-8/Compose:		en_BW.UTF-8
 en_US.UTF-8/Compose:		en_BZ.UTF-8
 en_US.UTF-8/Compose:		en_CA.UTF-8
+en_US.UTF-8/Compose:		en_DK.UTF-8
 en_US.UTF-8/Compose:		en_GB.UTF-8
+en_US.UTF-8/Compose:		en_HK.UTF-8
 en_US.UTF-8/Compose:		en_IE.UTF-8
+en_US.UTF-8/Compose:		en_IN.UTF-8
 en_US.UTF-8/Compose:		en_JM.UTF-8
 en_US.UTF-8/Compose:		en_NZ.UTF-8
+en_US.UTF-8/Compose:		en_PH.UTF-8
+en_US.UTF-8/Compose:		en_SG.UTF-8
 en_US.UTF-8/Compose:		en_TT.UTF-8
 en_US.UTF-8/Compose:		en_UK.UTF-8
 en_US.UTF-8/Compose:		en_US.UTF-8
 en_US.UTF-8/Compose:		en_ZA.UTF-8
-en_US.UTF-8/Compose:		eo_XX.UTF-8
+en_US.UTF-8/Compose:		en_ZW.UTF-8
+XCOMM Esperato (eo) is not supported by the GNU C Library and neither the
+XCOMM EO nor XX territories exist in ISO 3166.
+XCOMM en_US.UTF-8/Compose:	eo_EO.UTF-8
+XCOMM en_US.UTF-8/Compose:	eo_XX.UTF-8
 en_US.UTF-8/Compose:		es_AR.UTF-8
 en_US.UTF-8/Compose:		es_BO.UTF-8
 en_US.UTF-8/Compose:		es_CL.UTF-8
@@ -299,10 +377,12 @@
 en_US.UTF-8/Compose:		es_PR.UTF-8
 en_US.UTF-8/Compose:		es_PY.UTF-8
 en_US.UTF-8/Compose:		es_SV.UTF-8
+en_US.UTF-8/Compose:		es_US.UTF-8
 en_US.UTF-8/Compose:		es_UY.UTF-8
 en_US.UTF-8/Compose:		es_VE.UTF-8
 en_US.UTF-8/Compose:		et_EE.UTF-8
 en_US.UTF-8/Compose:		eu_ES.UTF-8
+en_US.UTF-8/Compose:		eu_FR.UTF-8
 en_US.UTF-8/Compose:		fa_IR.UTF-8
 en_US.UTF-8/Compose:		fi_FI.UTF-8
 en_US.UTF-8/Compose:		fo_FO.UTF-8
@@ -313,6 +393,8 @@
 en_US.UTF-8/Compose:		fr_LU.UTF-8
 en_US.UTF-8/Compose:		ga_IE.UTF-8
 en_US.UTF-8/Compose:		gd_GB.UTF-8
+en_US.UTF-8/Compose:       gez_ER.UTF-8
+en_US.UTF-8/Compose:       gez_ET.UTF-8
 en_US.UTF-8/Compose:		gl_ES.UTF-8
 en_US.UTF-8/Compose:		gu_IN.UTF-8
 en_US.UTF-8/Compose:		gv_GB.UTF-8
@@ -326,9 +408,12 @@
 en_US.UTF-8/Compose:		it_CH.UTF-8
 en_US.UTF-8/Compose:		it_IT.UTF-8
 en_US.UTF-8/Compose:		iu_CA.UTF-8
+en_US.UTF-8/Compose:       iw_IL.UTF-8
 en_US.UTF-8/Compose:		ja_JP.UTF-8
 en_US.UTF-8/Compose:		ka_GE.UTF-8
 en_US.UTF-8/Compose:		kl_GL.UTF-8
+en_US.UTF-8/Compose:		km_KH.UTF-8
+en_US.UTF-8/Compose:       kn_IN.UTF-8
 en_US.UTF-8/Compose:		ko_KR.UTF-8
 en_US.UTF-8/Compose:		kw_GB.UTF-8
 en_US.UTF-8/Compose:		ky_KG.UTF-8
@@ -337,18 +422,30 @@
 en_US.UTF-8/Compose:		lv_LV.UTF-8
 en_US.UTF-8/Compose:		mi_NZ.UTF-8
 en_US.UTF-8/Compose:		mk_MK.UTF-8
+en_US.UTF-8/Compose:       ml_IN.UTF-8
+en_US.UTF-8/Compose:       mn_MN.UTF-8
+en_US.UTF-8/Compose:       mr_IN.UTF-8
 en_US.UTF-8/Compose:		ms_MY.UTF-8
+en_US.UTF-8/Compose:       mt_MT.UTF-8
+en_US.UTF-8/Compose:       ne_NP.UTF-8
 en_US.UTF-8/Compose:		nl_BE.UTF-8
 en_US.UTF-8/Compose:		nl_NL.UTF-8
 en_US.UTF-8/Compose:		no_NO.UTF-8
 en_US.UTF-8/Compose:		nb_NO.UTF-8
 en_US.UTF-8/Compose:		nr_ZA.UTF-8
 en_US.UTF-8/Compose:		nso_ZA.UTF-8
-en_US.UTF-8/Compose:		ny_NO.UTF-8
+en_US.UTF-8/Compose:       nn_NO.UTF-8
+XCOMM ny is the language code for Nyanja (spoken in Malawi), not Nynorsk.
+XCOMM See <URL: http://www.ethnologue.com/show_iso639.asp?code=nya >.
+XCOMM en_US.UTF-8/Compose: ny_NO.UTF-8
 en_US.UTF-8/Compose:		oc_FR.UTF-8
+en_US.UTF-8/Compose:       om_ET.UTF-8
 en_US.UTF-8/Compose:		pa_IN.UTF-8
-en_US.UTF-8/Compose:		pd_DE.UTF-8
-en_US.UTF-8/Compose:		pd_US.UTF-8
+XCOMM Presumably, "pd" is "Pennsylvania 'Dutch'", a U.S. dialect of German.
+XCOMM This is not an ISO language code defined for it, and the GNU C Library
+XCOMM doesn't support it.
+XCOMM en_US.UTF-8/Compose: pd_DE.UTF-8
+XCOMM en_US.UTF-8/Compose: pd_US.UTF-8
 en_US.UTF-8/Compose:		ph_PH.UTF-8
 en_US.UTF-8/Compose:		pl_PL.UTF-8
 en_US.UTF-8/Compose:		pp_AN.UTF-8
@@ -358,9 +455,13 @@
 en_US.UTF-8/Compose:		ru_RU.UTF-8
 en_US.UTF-8/Compose:		ru_UA.UTF-8
 en_US.UTF-8/Compose:		rw_RW.UTF-8
+en_US.UTF-8/Compose:       se_NO.UTF-8
+en_US.UTF-8/Compose:       sh_YU.UTF-8
+en_US.UTF-8/Compose:       sid_ET.UTF-8
 en_US.UTF-8/Compose:		si_LK.UTF-8
 en_US.UTF-8/Compose:		sk_SK.UTF-8
 en_US.UTF-8/Compose:		sl_SI.UTF-8
+en_US.UTF-8/Compose:       so_ET.UTF-8
 en_US.UTF-8/Compose:		sq_AL.UTF-8
 en_US.UTF-8/Compose:		sr_CS.UTF-8
 en_US.UTF-8/Compose:		sr_YU.UTF-8
@@ -369,8 +470,12 @@
 en_US.UTF-8/Compose:		sv_FI.UTF-8
 en_US.UTF-8/Compose:		sv_SE.UTF-8
 en_US.UTF-8/Compose:		ta_IN.UTF-8
+en_US.UTF-8/Compose:       te_IN.UTF-8
 en_US.UTF-8/Compose:		tg_TJ.UTF-8
 en_US.UTF-8/Compose:		th_TH.UTF-8
+en_US.UTF-8/Compose:       ti_ER.UTF-8
+en_US.UTF-8/Compose:       ti_ET.UTF-8
+en_US.UTF-8/Compose:       tig_ER.UTF-8
 en_US.UTF-8/Compose:		tl_PH.UTF-8
 en_US.UTF-8/Compose:		tn_ZA.UTF-8
 en_US.UTF-8/Compose:		tr_TR.UTF-8
Index: libx11/nls/locale.alias.pre
===================================================================
--- libx11.orig/nls/locale.alias.pre
+++ libx11/nls/locale.alias.pre
@@ -324,6 +324,7 @@
 en_US.ISO-8859-1:				en_US.ISO8859-1
 en_US.ISO_8859-1:				en_US.ISO8859-1
 en_US.iso885915:				en_US.ISO8859-15
+en_US.ISO-8859-15:				en_US.ISO8859-15
 en_US.8859-15:					en_US.ISO8859-15
 en_US.ISO8859-15@euro:				en_US.ISO8859-15
 en_US.utf8:					en_US.UTF-8
@@ -690,6 +691,10 @@
 kl_GL.ISO-8859-1:				kl_GL.ISO8859-1
 kl_GL.iso885915:				kl_GL.ISO8859-15
 kl_GL.utf8:					kl_GL.UTF-8
+km:						km_KH.UTF-8
+km_KH:						km_KH.UTF-8
+km_KH.utf8:					km_KH.UTF-8
+km_KH.UTF-8:					km_KH.UTF-8
 ko:						ko_KR.eucKR
 ko_KR:						ko_KR.eucKR
 KO_KR:						ko_KR.eucKR
@@ -858,7 +863,6 @@
 pt_BR.iso88591:					pt_BR.ISO8859-1
 pt_BR.ISO-8859-1:				pt_BR.ISO8859-1
 pt_BR.ISO_8859-1:				pt_BR.ISO8859-1
-pt_BR.iso885915:				pt_BR.ISO8859-15
 pt_BR.utf8:					pt_BR.UTF-8
 pt_PT:						pt_PT.ISO8859-1
 pt_PT.88591:					pt_PT.ISO8859-1
@@ -909,12 +913,12 @@
 XCOMM YU was the old ISO code for Yugoslavia (now CS for Serbia&Montenegro).
 XCOMM These lines should be deleted in the future.
 sh:						sr_CS.ISO8859-2
-sh_HR.iso88592:					hr_HR.ISO8859-2
 sh_SP:						sr_CS.ISO8859-2
 sh_YU:						sr_CS.ISO8859-2
 sh_YU.utf8:					sr_CS.UTF-8
 si:						si_LK.UTF-8
 si_LK:						si_LK.UTF-8
+sid_ET:                        sid_ET.UTF-8
 sk:						sk_SK.ISO8859-2
 sk_SK:						sk_SK.ISO8859-2
 sk_SK.iso88592:					sk_SK.ISO8859-2
@@ -1145,7 +1149,7 @@
 dutch:						nl_NL.ISO8859-1
 dutch.iso88591:					nl_BE.ISO8859-1
 eesti:						et_EE.ISO8859-1
-english.iso88591:				en_EN.ISO8859-1
+english.iso88591:				en_US.ISO8859-1
 estonian:					et_EE.ISO8859-1
 finnish:					fi_FI.ISO8859-1
 finnish.iso88591:				fi_FI.ISO8859-1
@@ -1158,8 +1162,8 @@
 german.iso88591:				de_CH.ISO8859-1
 greek:						el_GR.ISO8859-7
 greek.iso88597:					el_GR.ISO8859-7
-hebrew:						iw_IL.ISO8859-8
-hebrew.iso88598:				iw_IL.ISO8859-8
+hebrew:						he_IL.ISO8859-8
+hebrew.iso88598:				he_IL.ISO8859-8
 hrvatski:					hr_HR.ISO8859-2
 hungarian:					hu_HU.ISO8859-2
 icelandic.iso88591:				is_IS.ISO8859-1
@@ -1202,103 +1206,103 @@
 #ifdef WIN32
 XCOMM Microsoft Windows/NT 4.0 SP3
 XCOMM parser doesn't grok embedded spaces in locale name
-Afrikaans_South Africa.1252:			af_ZA.iso8859-1
-Arabic_Bahrain.1256:				ar_BH.iso8859-6
-Arabic_Algeria.1256:				ar_DZ.iso8859-6
-Arabic_Egypt.1256:				ar_EG.iso8859-6
-Arabic_Iraq.1256:				ar_IQ.iso8859-6
-Arabic_Jordan.1256:				ar_JO.iso8859-6
-Arabic_Kuwait.1256:				ar_KW.iso8859-6
-Arabic_Lebanon.1256:				ar_LB.iso8859-6
-Arabic_Libya.1256:				ar_LY.iso8859-6
-Arabic_Morocco.1256:				ar_MA.iso8859-6
-Arabic_Oman.1256:				ar_OM.iso8859-6
-Arabic_Qatar.1256:				ar_QA.iso8859-6
-Arabic_Saudi Arabia.1256:			ar_SA.iso8859-6
-Arabic_Tunisia.1256:				ar_TN.iso8859-6
-Arabic_Yemen.1256:				ar_YE.iso8859-6
-Belarusian_Belarus.1251:			be_BY.iso8859-5
-Bulgarian_Bulgaria.1251:			bg_BG.iso8859-5
-Catalan_Spain.1252:				ca_ES.iso8859-1
-Czech_Czech Republic.1250:			cs_CZ.iso8859-2
-Danish_Denmark.1252:				da_DK.iso8859-1
-German_Austria.1252:				de_AT.iso8859-1
-German_Switzerland.1252:			de_CH.iso8859-1
-German_Germany.1252:				de_DE.iso8859-1
-German_Liechtenstein.1252:			de_LI.iso8859-1
-German_Luxembourg.1252:				de_LU.iso8859-1
-Greek_Greece.1253:				el_GR.iso8859-7
-English_Australia.1252:				en_AU.iso8859-1
-English_Belize.1252:				en_BZ.iso8859-1
-English_Canada.1252:				en_CA.iso8859-1
-English_Ireland.1252:				en_IE.iso8859-1
-English_Jamaica.1252:				en_JM.iso8859-1
-English_New Zealand.1252:			en_NZ.iso8859-1
-English_Trinidad y Tobago.1252:			en_TT.iso8859-1
-English_United Kingdom.1252:			en_UK.iso8859-1
-English_United States.1252:			en_US.iso8859-1
-English_South Africa.1252:			en_ZA.iso8859-1
-Spanish_Argentina.1252:				es_AR.iso8859-1
-Spanish_Bolivia.1252:				es_BO.iso8859-1
-Spanish_Chile.1252:				es_CL.iso8859-1
-Spanish_Colombia.1252:				es_CO.iso8859-1
-Spanish_Costa Rica.1252:			es_CR.iso8859-1
-Spanish_Dominican Republic.1252:		es_DO.iso8859-1
-Spanish_Ecuador.1252:				es_EC.iso8859-1
-Spanish - Modern Sort_Spain.1252:		es_ES.iso8859-1
-Spanish - Traditional Sort_Spain.1252:		es_ES.iso8859-1
-Spanish_Guatemala.1252:				es_GT.iso8859-1
-Spanish_Honduras.1252:				es_HN.iso8859-1
-Spanish_Mexican.1252:				es_MX.iso8859-1
-Spanish_Nicaragua.1252:				es_NI.iso8859-1
-Spanish_Panama.1252:				es_PA.iso8859-1
-Spanish_Paraguay.1252:				es_PY.iso8859-1
-Spanish_Peru.1252:				es_PE.iso8859-1
-Spanish_Puerto Rico.1252:			es_PR.iso8859-1
-Spanish_El Salvador.1252:			es_SV.iso8859-1
-Spanish_Uruguay.1252:				es_UY.iso8859-1
-Spanish_Venezuela.1252:				es_VE.iso8859-1
-Estonian_Estonia.1257:				et_EE.iso8859-13
-Basque_Spain.1252:				eu_ES.iso8859-1
-Finnish_Finland.1252:				fi_FI.iso8859-1
-French_Belgium.1252:				fr_BE.iso8859-1
-French_Canada.1252:				fr_CA.iso8859-1
-French_Switzerland.1252:			fr_CH.iso8859-1
-French_France.1252:				fr_FR.iso8859-1
-French_Luxembourg.1252:				fr_LU.iso8859-1
-Faeroese_Faeroe Islands.1252:			fo_FO.iso8859-1
-Hebrew_Israel.1255:				he_IL.iso8859-8
-Croatian_Croatia.1250:				hr_HR.iso8859-2
-Hungarian_Hungary.1250:				hu_HU.iso8859-2
-Indonesian_Indonesia.1252:			id_ID.iso8859-1
-Icelandic_Iceland.1252:				is_IS.iso8859-1
-Italian_Switzerland.1252:			it_CH.iso8859-1
-Italian_Italy.1252:				it_IT.iso8859-1
-Japanese_Japan.932:				ja_JP.SJIS
-Korean_Korea.949:				ko_KR.EUC
-Lithuanian_Lithuania.1257:			lt_LT.iso8859-13
-Latvian_Latvia.1257:				lv_LV.iso8859-13
-Dutch_Belgium.1252:				nl_BE.iso8859-1
-Dutch_Netherlands.1252:				nl_NL.iso8859-1
-Norwegian (Nynorsk)_Norway.1252:		no_NO.iso8859-1
-Norwegian (Bokmål)_Norway.1252:			no_NO.iso8859-1
-Polish_Poland.1250:				pl_PL.iso8859-2
-Portuguese_Brazil.1252:				pt_BR.iso8859-1
-Portuguese_Portugal.1252:			pt_PT.iso8859-1
-Romanian_Romania.1250:				ro_RO.iso8859-2
-Russian_Russia.1251:				ru_RU.iso8859-5
-Slovak_Slovakia.1250:				sk_SK.iso8859-2
-Slovene_Slovenia.1250:				sl_SI.iso8859-2
-Albanian_Albania.1250:				sq_AL.iso8859-2
-Serbian (Latin)_Serbia.1250:			sr_CS.iso8859-2
-Swedish_Finland.1252:				sv_FI.iso8859-1
-Swedish_Sweden.1252:				sv_SE.iso8859-1
-Turkish_Turkey.1254:				tr_TR.iso8859-9
-Ukrainian_Ukraine.1251:				uk_UA.iso8859-5
-Chinese(PRC)_People's Republic of China.936:	zh_CN.EUC
-Chinese(PRC)_Hong Kong.950:			zh_HK.EUC
-Chinese(Singapore)_Signapore.936:		zh_SG.EUC
-Chinese(Taiwan)_Taiwan.950:			zh_TW.EUC
+Afrikaans_South Africa.1252:          af_ZA.ISO8859-1
+Arabic_Bahrain.1256:              ar_BH.ISO8859-6
+Arabic_Algeria.1256:              ar_DZ.ISO8859-6
+Arabic_Egypt.1256:                ar_EG.ISO8859-6
+Arabic_Iraq.1256:             ar_IQ.ISO8859-6
+Arabic_Jordan.1256:               ar_JO.ISO8859-6
+Arabic_Kuwait.1256:               ar_KW.ISO8859-6
+Arabic_Lebanon.1256:              ar_LB.ISO8859-6
+Arabic_Libya.1256:                ar_LY.ISO8859-6
+Arabic_Morocco.1256:              ar_MA.ISO8859-6
+Arabic_Oman.1256:             ar_OM.ISO8859-6
+Arabic_Qatar.1256:                ar_QA.ISO8859-6
+Arabic_Saudi Arabia.1256:         ar_SA.ISO8859-6
+Arabic_Tunisia.1256:              ar_TN.ISO8859-6
+Arabic_Yemen.1256:                ar_YE.ISO8859-6
+Belarusian_Belarus.1251:          be_BY.ISO8859-5
+Bulgarian_Bulgaria.1251:          bg_BG.ISO8859-5
+Catalan_Spain.1252:               ca_ES.ISO8859-1
+Czech_Czech Republic.1250:            cs_CZ.ISO8859-2
+Danish_Denmark.1252:              da_DK.ISO8859-1
+German_Austria.1252:              de_AT.ISO8859-1
+German_Switzerland.1252:          de_CH.ISO8859-1
+German_Germany.1252:              de_DE.ISO8859-1
+German_Liechtenstein.1252:            de_LI.ISO8859-1
+German_Luxembourg.1252:               de_LU.ISO8859-1
+Greek_Greece.1253:                el_GR.ISO8859-7
+English_Australia.1252:               en_AU.ISO8859-1
+English_Belize.1252:              en_BZ.ISO8859-1
+English_Canada.1252:              en_CA.ISO8859-1
+English_Ireland.1252:             en_IE.ISO8859-1
+English_Jamaica.1252:             en_JM.ISO8859-1
+English_New Zealand.1252:         en_NZ.ISO8859-1
+English_Trinidad y Tobago.1252:           en_TT.ISO8859-1
+English_United Kingdom.1252:          en_UK.ISO8859-1
+English_United States.1252:           en_US.ISO8859-1
+English_South Africa.1252:            en_ZA.ISO8859-1
+Spanish_Argentina.1252:               es_AR.ISO8859-1
+Spanish_Bolivia.1252:             es_BO.ISO8859-1
+Spanish_Chile.1252:               es_CL.ISO8859-1
+Spanish_Colombia.1252:                es_CO.ISO8859-1
+Spanish_Costa Rica.1252:          es_CR.ISO8859-1
+Spanish_Dominican Republic.1252:      es_DO.ISO8859-1
+Spanish_Ecuador.1252:             es_EC.ISO8859-1
+Spanish - Modern Sort_Spain.1252:     es_ES.ISO8859-1
+Spanish - Traditional Sort_Spain.1252:        es_ES.ISO8859-1
+Spanish_Guatemala.1252:               es_GT.ISO8859-1
+Spanish_Honduras.1252:                es_HN.ISO8859-1
+Spanish_Mexican.1252:             es_MX.ISO8859-1
+Spanish_Nicaragua.1252:               es_NI.ISO8859-1
+Spanish_Panama.1252:              es_PA.ISO8859-1
+Spanish_Paraguay.1252:                es_PY.ISO8859-1
+Spanish_Peru.1252:                es_PE.ISO8859-1
+Spanish_Puerto Rico.1252:         es_PR.ISO8859-1
+Spanish_El Salvador.1252:         es_SV.ISO8859-1
+Spanish_Uruguay.1252:             es_UY.ISO8859-1
+Spanish_Venezuela.1252:               es_VE.ISO8859-1
+Estonian_Estonia.1257:                et_EE.ISO8859-13
+Basque_Spain.1252:                eu_ES.ISO8859-1
+Finnish_Finland.1252:             fi_FI.ISO8859-1
+French_Belgium.1252:              fr_BE.ISO8859-1
+French_Canada.1252:               fr_CA.ISO8859-1
+French_Switzerland.1252:          fr_CH.ISO8859-1
+French_France.1252:               fr_FR.ISO8859-1
+French_Luxembourg.1252:               fr_LU.ISO8859-1
+Faeroese_Faeroe Islands.1252:         fo_FO.ISO8859-1
+Hebrew_Israel.1255:               he_IL.ISO8859-8
+Croatian_Croatia.1250:                hr_HR.ISO8859-2
+Hungarian_Hungary.1250:               hu_HU.ISO8859-2
+Indonesian_Indonesia.1252:            id_ID.ISO8859-1
+Icelandic_Iceland.1252:               is_IS.ISO8859-1
+Italian_Switzerland.1252:         it_CH.ISO8859-1
+Italian_Italy.1252:               it_IT.ISO8859-1
+Japanese_Japan.932:               ja_JP.SJIS
+Korean_Korea.949:             ko_KR.eucKR
+Lithuanian_Lithuania.1257:            lt_LT.ISO8859-13
+Latvian_Latvia.1257:              lv_LV.ISO8859-13
+Dutch_Belgium.1252:               nl_BE.ISO8859-1
+Dutch_Netherlands.1252:               nl_NL.ISO8859-1
+Norwegian (Nynorsk)_Norway.1252:      no_NO.ISO8859-1
+Norwegian (Bokmål)_Norway.1252:           no_NO.ISO8859-1
+Polish_Poland.1250:               pl_PL.ISO8859-2
+Portuguese_Brazil.1252:               pt_BR.ISO8859-1
+Portuguese_Portugal.1252:         pt_PT.ISO8859-1
+Romanian_Romania.1250:                ro_RO.ISO8859-2
+Russian_Russia.1251:              ru_RU.ISO8859-5
+Slovak_Slovakia.1250:             sk_SK.ISO8859-2
+Slovene_Slovenia.1250:                sl_SI.ISO8859-2
+Albanian_Albania.1250:                sq_AL.ISO8859-2
+Serbian (Latin)_Serbia.1250:          sr_YU.ISO8859-2
+Swedish_Finland.1252:             sv_FI.ISO8859-1
+Swedish_Sweden.1252:              sv_SE.ISO8859-1
+Turkish_Turkey.1254:              tr_TR.ISO8859-9
+Ukrainian_Ukraine.1251:               uk_UA.ISO8859-5
+Chinese(PRC)_People's Republic of China.936:  zh_CN.eucCN
+Chinese(PRC)_Hong Kong.950:           zh_TW.eucTW
+Chinese(Singapore)_Signapore.936:     zh_CN.eucCN
+Chinese(Taiwan)_Taiwan.950:           zh_TW.eucTW
 #endif
 XCOMM Other miscellaneous locale names
 ISO8859-1:					en_US.ISO8859-1
Index: libx11/nls/locale.dir.pre
===================================================================
--- libx11.orig/nls/locale.dir.pre
+++ libx11/nls/locale.dir.pre
@@ -9,8 +9,13 @@
 XCOMM
 
 C/XLC_LOCALE:				C
+XCOMM a3 is not an ISO 639 language code, but in Cyrillic, "Z" looks like "3".
+koi8-c/XLC_LOCALE:			a3_AZ.KOI8-C
+iso8859-1/XLC_LOCALE:			aa_DJ.ISO8859-1
 iso8859-1/XLC_LOCALE:			af_ZA.ISO8859-1
+iso8859-15/XLC_LOCALE:			an_ES.ISO8859-15
 iso8859-6/XLC_LOCALE:			ar_AA.ISO8859-6
+iso8859-6/XLC_LOCALE:			ar_AE.ISO8859-6
 iso8859-6/XLC_LOCALE:			ar_BH.ISO8859-6
 iso8859-6/XLC_LOCALE:			ar_DZ.ISO8859-6
 iso8859-6/XLC_LOCALE:			ar_EG.ISO8859-6
@@ -38,13 +43,13 @@
 iso8859-1/XLC_LOCALE:			br_FR.ISO8859-1
 iso8859-14/XLC_LOCALE:			br_FR.ISO8859-14
 iso8859-15/XLC_LOCALE:			br_FR.ISO8859-15
+iso8859-2/XLC_LOCALE:			bs_BA.ISO8859-2
 iso8859-1/XLC_LOCALE:			ca_ES.ISO8859-1
 iso8859-15/XLC_LOCALE:			ca_ES.ISO8859-15
 iso8859-2/XLC_LOCALE:			cs_CZ.ISO8859-2
 iso8859-1/XLC_LOCALE:			cy_GB.ISO8859-1
 iso8859-14/XLC_LOCALE:			cy_GB.ISO8859-14
 iso8859-15/XLC_LOCALE:			cy_GB.ISO8859-15
-iso8859-2/XLC_LOCALE:			cz_CZ.ISO8859-2
 iso8859-1/XLC_LOCALE:			da_DK.ISO8859-1
 iso8859-15/XLC_LOCALE:			da_DK.ISO8859-15
 iso8859-1/XLC_LOCALE:			de_AT.ISO8859-1
@@ -56,20 +61,24 @@
 iso8859-1/XLC_LOCALE:			de_DE.ISO8859-1
 iso8859-15/XLC_LOCALE:			de_DE.ISO8859-15
 iso8859-1/XLC_LOCALE:			de_LI.ISO8859-1
+iso8859-15/XLC_LOCALE:			de_LI.ISO8859-15
 iso8859-1/XLC_LOCALE:			de_LU.ISO8859-1
 iso8859-15/XLC_LOCALE:			de_LU.ISO8859-15
-iso8859-7/XLC_LOCALE: 			el_GR.ISO8859-7
+iso8859-7/XLC_LOCALE:			el_GR.ISO8859-7
 iso8859-15/XLC_LOCALE:			el_GR.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_AU.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_BE.ISO8859-1
+iso8859-15/XLC_LOCALE:			en_BE.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_BZ.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_BW.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_CA.ISO8859-1
+iso8859-1/XLC_LOCALE:			en_DK.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_GB.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_GB.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_HK.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_IE.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_IE.ISO8859-15
+iso8859-1/XLC_LOCALE:			en_IN.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_IN.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_JM.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_NZ.ISO8859-1
@@ -82,8 +91,11 @@
 iso8859-15/XLC_LOCALE:			en_US.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_ZA.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_ZA.ISO8859-15
-iso8859-3/XLC_LOCALE:			eo_EO.ISO8859-3
-iso8859-3/XLC_LOCALE:			eo_XX.ISO8859-3
+iso8859-1/XLC_LOCALE:			en_ZW.ISO8859-1
+XCOMM Esperato (eo) is not supported by the GNU C Library and neither the
+XCOMM EO nor XX territories exist in ISO 3166.
+XCOMM iso8859-3/XLC_LOCALE:			eo_EO.ISO8859-3
+XCOMM iso8859-3/XLC_LOCALE:			eo_XX.ISO8859-3
 iso8859-1/XLC_LOCALE:			es_AR.ISO8859-1
 iso8859-1/XLC_LOCALE:			es_BO.ISO8859-1
 iso8859-1/XLC_LOCALE:			es_CL.ISO8859-1
@@ -111,6 +123,8 @@
 iso8859-15/XLC_LOCALE:			et_EE.ISO8859-15
 iso8859-1/XLC_LOCALE:			eu_ES.ISO8859-1
 iso8859-15/XLC_LOCALE:			eu_ES.ISO8859-15
+iso8859-1/XLC_LOCALE:			eu_FR.ISO8859-1
+iso8859-15/XLC_LOCALE:			eu_FR.ISO8859-15
 isiri-3342/XLC_LOCALE:			fa_IR.ISIRI-3342
 iso8859-1/XLC_LOCALE:			fi_FI.ISO8859-1
 iso8859-15/XLC_LOCALE:			fi_FI.ISO8859-15
@@ -159,6 +173,7 @@
 iso8859-1/XLC_LOCALE:			kw_GB.ISO8859-1
 iso8859-14/XLC_LOCALE:			kw_GB.ISO8859-14
 iso8859-15/XLC_LOCALE:			kw_GB.ISO8859-15
+iso8859-10/XLC_LOCALE:			lg_UG.ISO8859-10
 ibm-cp1133/XLC_LOCALE:			lo_LA.IBM-CP1133
 mulelao-1/XLC_LOCALE:			lo_LA.MULELAO-1
 iso8859-4/XLC_LOCALE:			lt_LT.ISO8859-4
@@ -186,8 +201,12 @@
 iso8859-1/XLC_LOCALE:			ny_NO.ISO8859-15
 iso8859-1/XLC_LOCALE:			oc_FR.ISO8859-1
 iso8859-15/XLC_LOCALE:			oc_FR.ISO8859-15
-iso8859-1/XLC_LOCALE:			pd_DE.ISO8859-1
-iso8859-1/XLC_LOCALE:			ph_PH.ISO8859-1
+iso8859-1/XLC_LOCALE:			om_KE.ISO8859-1
+XCOMM Presumably, "pd" is "Pennsylvania 'Dutch'", a U.S. dialect of German.
+XCOMM This is not an ISO language code defined for it, and the GNU C Library
+XCOMM doesn't support it.
+XCOMM iso8859-1/XLC_LOCALE:			pd_DE.ISO8859-1
+Iso8859-1/XLC_LOCALE:			ph_PH.ISO8859-1
 iso8859-2/XLC_LOCALE:			pl_PL.ISO8859-2
 iso8859-1/XLC_LOCALE:			pp_AN.ISO8859-1
 iso8859-1/XLC_LOCALE:			pt_BR.ISO8859-1
@@ -197,11 +216,15 @@
 iso8859-5/XLC_LOCALE: 			ru_RU.ISO8859-5
 microsoft-cp1251/XLC_LOCALE: 		ru_RU.CP1251
 koi8-r/XLC_LOCALE: 			ru_RU.KOI8-R
+microsoft-cp1251/XLC_LOCALE:       ru_UA.CP1251
 koi8-u/XLC_LOCALE:			ru_UA.KOI8-U
 iso8859-1/XLC_LOCALE:           rw_RW.ISO8859-1
 iso8859-2/XLC_LOCALE: 			sh_YU.ISO8859-2
 iso8859-2/XLC_LOCALE: 			sk_SK.ISO8859-2
 iso8859-2/XLC_LOCALE: 			sl_SI.ISO8859-2
+iso8859-1/XLC_LOCALE:          so_DJ.ISO8859-1
+iso8859-1/XLC_LOCALE:          so_KE.ISO8859-1
+iso8859-1/XLC_LOCALE:          so_SO.ISO8859-1
 iso8859-5/XLC_LOCALE: 			sp_YU.ISO8859-5
 iso8859-2/XLC_LOCALE: 			sq_AL.ISO8859-2
 iso8859-2/XLC_LOCALE:			sr_CS.ISO8859-2
@@ -218,8 +241,8 @@
 iso8859-15/XLC_LOCALE:			sv_SE.ISO8859-15
 tscii-0/XLC_LOCALE:			ta_IN.TSCII-0
 koi8-c/XLC_LOCALE:			tg_TJ.KOI8-C
-th_TH/XLC_LOCALE:			th_TH.TIS620
 iso8859-11/XLC_LOCALE:			th_TH.ISO8859-11
+th_TH/XLC_LOCALE:			th_TH.TIS620
 iso8859-1/XLC_LOCALE: 			tl_PH.ISO8859-1
 iso8859-15/XLC_LOCALE:			tn_ZA.ISO8859-15
 iso8859-9/XLC_LOCALE:			tr_TR.ISO8859-9
@@ -230,6 +253,7 @@
 microsoft-cp1251/XLC_LOCALE: 		uk_UA.CP1251
 koi8-u/XLC_LOCALE:			uk_UA.KOI8-U
 microsoft-cp1256/XLC_LOCALE:		ur_PK.CP1256
+iso8859-1/XLC_LOCALE:          uz_UZ.ISO8859-1
 vi_VN.tcvn/XLC_LOCALE:			vi_VN.TCVN
 vi_VN.viscii/XLC_LOCALE:		vi_VN.VISCII
 iso8859-1/XLC_LOCALE:			wa_BE.ISO8859-1
@@ -245,10 +269,17 @@
 zh_HK.big5hkscs/XLC_LOCALE:		zh_HK.Big5-HKSCS
 zh_HK.big5hkscs/XLC_LOCALE:		zh_HK.big5hkscs
 zh_HK.big5hkscs/XLC_LOCALE:		zh_HK.Big5HKSCS
+zh_CN/XLC_LOCALE:          zh_SG.gb2312
+zh_CN.gbk/XLC_LOCALE:          zh_SG.gbk
 zh_TW.big5/XLC_LOCALE:			zh_TW.big5
 zh_TW/XLC_LOCALE:			zh_TW.eucTW
 iso8859-1/XLC_LOCALE:			zu_ZA.ISO8859-1
 XCOMM Note: The UTF-8 locales don't work correctly yet. Work in progress.
+XCOMM a3 is not an ISO 639 language code, but in Cyrillic, "Z" looks like
+XCOMM "3".
+en_US.UTF-8/XLC_LOCALE:            a3_AZ.UTF-8
+en_US.UTF-8/XLC_LOCALE:            aa_ER.UTF-8
+en_US.UTF-8/XLC_LOCALE:            aa_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			af_ZA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			am_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ar_AA.UTF-8
@@ -256,6 +287,7 @@
 en_US.UTF-8/XLC_LOCALE:			ar_BH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ar_DZ.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ar_EG.UTF-8
+en_US.UTF-8/XLC_LOCALE:			ar_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ar_IQ.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ar_JO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ar_KW.UTF-8
@@ -271,12 +303,12 @@
 en_US.UTF-8/XLC_LOCALE:			ar_TN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ar_YE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			az_AZ.UTF-8
-en_US.UTF-8/XLC_LOCALE:			a3_AZ.UTF-8
 en_US.UTF-8/XLC_LOCALE:			be_BY.UTF-8
 en_US.UTF-8/XLC_LOCALE:			bg_BG.UTF-8
 en_US.UTF-8/XLC_LOCALE:			bn_BD.UTF-8
 en_US.UTF-8/XLC_LOCALE:			bn_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			br_FR.UTF-8
+en_US.UTF-8/XLC_LOCALE:            byn_ER.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ca_ES.UTF-8
 en_US.UTF-8/XLC_LOCALE:			cs_CZ.UTF-8
 en_US.UTF-8/XLC_LOCALE:			cy_GB.UTF-8
@@ -291,18 +323,27 @@
 en_US.UTF-8/XLC_LOCALE:			el_GR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_AU.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_BE.UTF-8
+en_US.UTF-8/XLC_LOCALE:			en_BW.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_BZ.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_CA.UTF-8
+en_US.UTF-8/XLC_LOCALE:			en_DK.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_GB.UTF-8
+en_US.UTF-8/XLC_LOCALE:			en_HK.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_IE.UTF-8
+en_US.UTF-8/XLC_LOCALE:			en_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_JM.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_NZ.UTF-8
+en_US.UTF-8/XLC_LOCALE:			en_PH.UTF-8
+en_US.UTF-8/XLC_LOCALE:			en_SG.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_TT.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_UK.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_US.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_ZA.UTF-8
-en_US.UTF-8/XLC_LOCALE:			eo_EO.UTF-8
-en_US.UTF-8/XLC_LOCALE:			eo_XX.UTF-8
+en_US.UTF-8/XLC_LOCALE:			en_ZW.UTF-8
+XCOMM Esperato (eo) is not supported by the GNU C Library and neither the
+XCOMM EO nor XX territories exist in ISO 3166.
+XCOMM en_US.UTF-8/XLC_LOCALE:			eo_EO.UTF-8
+XCOMM en_US.UTF-8/XLC_LOCALE:			eo_XX.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_AR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_BO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_CL.UTF-8
@@ -320,10 +361,12 @@
 en_US.UTF-8/XLC_LOCALE:			es_PR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_PY.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_SV.UTF-8
+en_US.UTF-8/XLC_LOCALE:			es_US.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_UY.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_VE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			et_EE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			eu_ES.UTF-8
+en_US.UTF-8/XLC_LOCALE:			eu_FR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			fa_IR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			fi_FI.UTF-8
 en_US.UTF-8/XLC_LOCALE:			fo_FO.UTF-8
@@ -334,6 +377,8 @@
 en_US.UTF-8/XLC_LOCALE:			fr_LU.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ga_IE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			gd_GB.UTF-8
+en_US.UTF-8/XLC_LOCALE:            gez_ER.UTF-8
+en_US.UTF-8/XLC_LOCALE:            gez_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			gl_ES.UTF-8
 en_US.UTF-8/XLC_LOCALE:			gu_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			gv_GB.UTF-8
@@ -347,9 +392,12 @@
 en_US.UTF-8/XLC_LOCALE:			it_CH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			it_IT.UTF-8
 en_US.UTF-8/XLC_LOCALE:			iu_CA.UTF-8
+en_US.UTF-8/XLC_LOCALE:			iw_IL.UTF-8
 ja_JP.UTF-8/XLC_LOCALE:			ja_JP.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ka_GE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			kl_GL.UTF-8
+en_US.UTF-8/XLC_LOCALE:			km_KH.UTF-8
+en_US.UTF-8/XLC_LOCALE:			kn_IN.UTF-8
 ko_KR.UTF-8/XLC_LOCALE:			ko_KR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			kw_GB.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ky_KG.UTF-8
@@ -358,20 +406,30 @@
 en_US.UTF-8/XLC_LOCALE:			lv_LV.UTF-8
 en_US.UTF-8/XLC_LOCALE:			mi_NZ.UTF-8
 en_US.UTF-8/XLC_LOCALE:			mk_MK.UTF-8
+en_US.UTF-8/XLC_LOCALE:            ml_IN.UTF-8
+en_US.UTF-8/XLC_LOCALE:            mn_MN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			mr_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ms_MY.UTF-8
+en_US.UTF-8/XLC_LOCALE:            mt_MT.UTF-8
+en_US.UTF-8/XLC_LOCALE:            ne_NP.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nl_BE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nl_NL.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nb_NO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nn_NO.UTF-8
+XCOMM ny is the language code for Nyanja (spoken in Malawi), not Nynorsk.
+XCOMM See <URL: http://www.ethnologue.com/show_iso639.asp?code=nya >.
+XCOMM en_US.UTF-8/XLC_LOCALE:          ny_NO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			no_NO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nr_ZA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nso_ZA.UTF-8
-en_US.UTF-8/XLC_LOCALE:			ny_NO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			oc_FR.UTF-8
+en_US.UTF-8/XLC_LOCALE:            om_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			pa_IN.UTF-8
-en_US.UTF-8/XLC_LOCALE:			pd_DE.UTF-8
-en_US.UTF-8/XLC_LOCALE:			pd_US.UTF-8
+XCOMM Presumably, "pd" is "Pennsylvania 'Dutch'", a U.S. dialect of German.
+XCOMM This is not an ISO language code defined for it, and the GNU C Library
+XCOMM doesn't support it.
+XCOMM en_US.UTF-8/XLC_LOCALE:          pd_DE.UTF-8
+XCOMM en_US.UTF-8/XLC_LOCALE:          pd_US.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ph_PH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			pl_PL.UTF-8
 en_US.UTF-8/XLC_LOCALE:			pp_AN.UTF-8
@@ -382,10 +440,12 @@
 en_US.UTF-8/XLC_LOCALE:			ru_UA.UTF-8
 en_US.UTF-8/XLC_LOCALE:         rw_RW.UTF-8
 en_US.UTF-8/XLC_LOCALE:			se_NO.UTF-8
+en_US.UTF-8/XLC_LOCALE:            sid_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sh_YU.UTF-8
 en_US.UTF-8/XLC_LOCALE:			si_LK.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sk_SK.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sl_SI.UTF-8
+en_US.UTF-8/XLC_LOCALE:            so_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sq_AL.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sr_CS.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sr_YU.UTF-8
@@ -394,10 +454,12 @@
 en_US.UTF-8/XLC_LOCALE:			sv_FI.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sv_SE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ta_IN.UTF-8
+en_US.UTF-8/XLC_LOCALE:            te_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			tg_TJ.UTF-8
 th_TH.UTF-8/XLC_LOCALE:			th_TH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ti_ER.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ti_ET.UTF-8
+en_US.UTF-8/XLC_LOCALE:            tig_ER.UTF-8
 en_US.UTF-8/XLC_LOCALE:			tl_PH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			tn_ZA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			tr_TR.UTF-8
